public/components/incontext_insight/generate_popover_body.tsx

```diff
@@ -20,21 +19,27 @@ import {
   EuiTitle,
 } from '@elastic/eui';
 import { useEffectOnce } from 'react-use';
+import { MessageActions } from '../../tabs/chat/messages/message_action';
 import { IncontextInsight as IncontextInsightInput } from '../../types';
 import { getNotifications } from '../../services';
 import { HttpSetup } from '../../../../../src/core/public';
 import { SUMMARY_ASSISTANT_API } from '../../../common/constants/llm';
 import shiny_sparkle from '../../assets/shiny_sparkle.svg';
+import { UsageCollectionSetup } from '../../../../../src/plugins/usage_collection/public';
+import { reportMetric } from '../../utils/report_metric';
 
 export const GeneratePopoverBody: React.FC<{
   incontextInsight: IncontextInsightInput;
   httpSetup?: HttpSetup;
+  usageCollection?: UsageCollectionSetup;
   closePopover: () => void;
-}> = ({ incontextInsight, httpSetup, closePopover }) => {
+}> = ({ incontextInsight, httpSetup, usageCollection, closePopover }) => {
   const [summary, setSummary] = useState('');
   const [insight, setInsight] = useState('');
   const [insightAvailable, setInsightAvailable] = useState(false);
   const [showInsight, setShowInsight] = useState(false);
+  const metricAppName = 'alertSumm';
```

Comment:
Does metric name have length restriction? If this is the case, you might need to add some protection check in util function.

Reply:
Actually I do not really know any specific restriction

Reply:
should we use full name `alertSummary`?